pyodide-build/pyodide_build/buildall.py

```diff
@@ -554,17 +540,38 @@ def generate_repodata(
                     Path(output_dir, pkg.unvendored_tests.name)
                 ),
             }
-            package_data["packages"][name.lower() + "-tests"] = pkg_entry
+            packages[name.lower() + "-tests"] = pkg_entry
 
     # Workaround for circular dependency between soupsieve and beautifulsoup4
     # TODO: FIXME!!
-    if "soupsieve" in package_data["packages"]:
-        package_data["packages"]["soupsieve"]["depends"].append("beautifulsoup4")
+    if "soupsieve" in packages:
+        packages["soupsieve"]["depends"].append("beautifulsoup4")
 
-    # re-order packages by name
-    package_data["packages"] = dict(sorted(package_data["packages"].items()))
+    # sort packages by name
+    packages = dict(sorted(packages.items()))
+    return packages
 
-    return package_data
+
+def generate_repodata(
+    output_dir: Path, pkg_map: dict[str, BasePackage]
+) -> dict[str, dict[str, Any]]:
+    """Generate the package.json file"""
+
+    import sys
+
+    sys.path.append(str(common.get_pyodide_root() / "src/py"))
+    from pyodide import __version__
+
+    # Build package.json data.
+    [platform, _, arch] = common.platform().rpartition("_")
+    info = {
+        "arch": arch,
+        "platform": platform,
+        "version": __version__,
+        "python": sys.version.partition(" ")[0],
```

Comment:
Maybe use [get_pyversion()](https://github.com/pyodide/pyodide/blob/060f1f4ba6cb4e1f8afbb6a474a70756415364aa/pyodide-build/pyodide_build/common.py#L150) to get the exact Pyodide Python version?